src/ImageLoaders/Pef/PefFile.cs

```diff
@@ -0,0 +1,104 @@
+using Reko.Core;
+using Reko.Core.Types;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Reko.ImageLoaders.Pef
+{
+    public class PefFile
+    {
+        public readonly PefContainer Container;
+        public PefLoaderInfo LoaderInfo;
+
+        private PefImageSegment[] pefSegments;
+
+        private PefFile(PefContainer container, PefImageSegment[] imageSegments)
+        {
+            this.Container = container;
+            this.pefSegments = imageSegments;
+        }
+
+        private PefLoaderInfo ProcessLoaderSection(PefImageSegment loaderSegment)
+        {
+            var pefLoaderSegment = PefLoaderSegment.Load(loaderSegment);
+            var loaderInfo = PefLoaderInfo.Load(pefLoaderSegment);
+            return loaderInfo;
+        }
+
+        private Address GetAddress(int sectionIndex, uint sectionOffset)
+        {
+            switch (sectionIndex)
+            {
+            // offset is the absolute address value
+            case -2: return new Address32(sectionOffset);
+            // offset is the imported symbol index
+            case -3: throw new NotImplementedException();
+            }
+            var baseAddr = pefSegments[sectionIndex].Segment.Address;
+            return baseAddr + sectionOffset;
+        }
+
+        public IEnumerable<ImageSymbol> GetEntryPoints(IProcessorArchitecture arch)
+        {
+            var infoHdr = LoaderInfo.Loader.InfoHeader;
+            if (infoHdr.initSection > -1)
+            {
+                yield return ImageSymbol.Procedure(
+                    arch,
+                    GetAddress(infoHdr.initSection, infoHdr.initOffset),
+                    "init"
+                );
+            }
+
+            if(infoHdr.mainSection > -1)
+            {
+                yield return ImageSymbol.Procedure(
+                    arch,
+                    GetAddress(infoHdr.mainSection, infoHdr.mainOffset),
+                    "main"
```

Comment:
Does MacOS differentiate between `_startup` and `main` the way ELF binaries do?

Reply:
It looks like, but i'm not sure which is the correct name for the startup symbol (`init` is made up)